src/sync.cpp

```diff
@@ -548,19 +548,28 @@ SyncConfigBag::SyncConfigBag(DbAccess& dbaccess, FileSystemAccess& fsaccess, Prn
 
 void SyncConfigBag::insert(const SyncConfig& syncConfig)
 {
+    auto insertOrUpdate = [this](const uint32_t id, const SyncConfig& syncConfig)
+    {
+        std::string data;
+        const_cast<SyncConfig&>(syncConfig).serialize(&data);
+        DBTableTransactionCommitter committer{mTable.get()};
+        if (!mTable->put(id, &data)) // put either inserts or updates
+        {
+            LOG_err << "Incomplete database put at id: " << mTable->nextid;
+            assert(false);
+            mTable->abort();
+            return false;
+        }
+        return true;
+    };
+
     auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
-    if (syncConfigPair == mSyncConfigs.end())
+    if (syncConfigPair == mSyncConfigs.end()) // syncConfig is new
```

Comment:
`syncConfigPair` is an iterator. It should be named `syncConfigIt` or alike, but not `...Pair`, which is misleading. I'll change it.